app/javascript/src/Navigation/api_selector.jsx

```diff
@@ -0,0 +1,77 @@
+// TODO: @flow
+// TODO: test
+import React, { Component } from 'react'
+import { render } from 'react-dom'
+
+const apiPathRoot = apiId => `/apiconfig/services/${apiId}/metrics`
+
+const Suggestions = (props) => {
+  const { apis } = props
+  const options = apis.map((api, index) => (
+    <li key={index} className="PopNavigation-listItem">
+      <a className="PopNavigation-link" href={`${apiPathRoot(api.service.id)}`}>{api.service.name}</a>
+    </li>
+  ))
+  return <ul className="PopNavigation-results">{options}</ul>
+}
+
+const searchBoxTitle = (api, controllerName) => {
+  const title = (api && controllerName !== 'dashboards') ? `API: ${api.service.name.toUpperCase()}` : 'Jumpt to an API'
+  return (<span> {title} <i className='fa fa-chevron-down'></i></span>)
+}
+
+class ApiSearch extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      ...props,
```

Comment:
This one is an anti-pattern, is it required?

Reply:
Defining the initial state with the props. I can see why `displayedApis: props.apis` but the not the whole set.

Reply:
Later you use `currentApi` and `controllerName` in https://github.com/3scale/porta/pull/31/files#diff-482a43efac952f091f24257e7abe5c7fR43 
I could destructure it before like `{ apis, currentApi, controllerName } = props` to make it explicit if it helps

Reply:
true, you are right 👍